registration_form/registration_form.js

```diff
@@ -0,0 +1,48 @@
+function validUname(uname) {
+    let error = 0;
+    var length = uname.length;
+    var validChars = /\w{3,10}/;
+    document.getElementById('username_error').innerHTML = '';
+    if (length < 3 || length > 10 || !validChars.test(uname)) {
+        error++;
+        document.getElementById('username_error').innerHTML = "<em style='color:red'>Please enter valid username</em>";
+    }
+    return error;
+}
+function validPass(pass) {
+    let error = 0;
+    var length = pass.length;
+    var validChars = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]{6,}$/;
+    document.getElementById('password_error').innerHTML = '';
+    if (length < 6 || !validChars.test(pass)) {
+        error++;
+        document.getElementById('password_error').innerHTML = "<em style='color:red'>Please enter valid password</em>";
+    }
+    return error;
+}
+function validPassConfirm(passConfirm, pass) {
+    let error = 0;
+    document.getElementById('passwordConfirm_error').innerHTML = '';
+    if (pass.localeCompare(passConfirm) !== 0) {
```

Comment:
Are you sure you want to rely only on validPass() to check if the chars are complying to the requirements for valid characters?